File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/RedisService.java
package org.javamaster.invocationlab.admin.service;

import org.javamaster.invocationlab.admin.model.redis.CommonRedisVo;
import org.javamaster.invocationlab.admin.model.redis.ConnectionVo;
import org.javamaster.invocationlab.admin.model.redis.Tree;
import org.javamaster.invocationlab.admin.model.redis.ValueVo;

import java.util.List;

/**
 * @author yudong
 */
public interface RedisService {
    String saveConnect(ConnectionVo connectionVoReq) throws Exception;

    List<Tree> listDb(String connectId) throws Exception;

    String pingConnect(ConnectionVo connectionVoReq) throws Exception;

    List<ConnectionVo> listConnects() throws Exception;

    List<Tree> listKeys(String connectId, Integer redisDbIndex, String pattern) throws Exception;

    ValueVo getValue(CommonRedisVo commonRedisVo) throws Exception;

    ValueVo renameKey(CommonRedisVo commonRedisVo) throws Exception;

    String delKey(CommonRedisVo commonRedisVo) throws Exception;

    ValueVo saveValue(CommonRedisVo commonRedisVo) throws Exception;

    String setNewTtl(CommonRedisVo commonRedisVo) throws Exception;

    ValueVo addKey(CommonRedisVo commonRedisVo) throws Exception;

    String delField(CommonRedisVo commonRedisVo) throws Exception;

    String addField(CommonRedisVo commonRedisVo) throws Exception;
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/RedisService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.model.redis.CommonRedisVo
import org.javamaster.invocationlab.admin.model.redis.ConnectionVo
import org.javamaster.invocationlab.admin.model.redis.Tree
import org.javamaster.invocationlab.admin.model.redis.ValueVo

/**
 * @author yudong
 */
interface RedisService {

    fun saveConnect(connectionVoReq: ConnectionVo): String

    fun listDb(connectId: String): List<Tree>

    fun pingConnect(connectionVoReq: ConnectionVo): String

    fun listConnects(): List<ConnectionVo>

    fun listKeys(connectId: String, redisDbIndex: Int, pattern: String): List<Tree>

    fun getValue(commonRedisVo: CommonRedisVo): ValueVo

    fun renameKey(commonRedisVo: CommonRedisVo): ValueVo

    fun delKey(commonRedisVo: CommonRedisVo): String

    fun saveValue(commonRedisVo: CommonRedisVo): ValueVo

    fun setNewTtl(commonRedisVo: CommonRedisVo): String

    fun addKey(commonRedisVo: CommonRedisVo): ValueVo

    fun delField(commonRedisVo: CommonRedisVo): String

    fun addField(commonRedisVo: CommonRedisVo): String
}
